includes/admin/class-wc-rest-payments-accounts-controller.php

```diff
@@ -35,11 +35,44 @@ public function register_routes() {
 	}
 
 	/**
-	 * Get accounts details via API.
+	 * Get account details via API.
 	 *
-	 * @param WP_REST_Request $request Full data about the request.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
 	 */
 	public function get_account_data( $request ) {
-		return WC_Payments::get_account_service()->get_cached_account_data();
+		// Inject the API client supplied into the controller, in order to enable testing.
+		$account_service     = WC_Payments::get_account_service();
+		$original_api_client = $account_service->set_payments_api_client( $this->api_client );
+
+		// Fetch account and restore original API client, in order to reset the core state.
+		$account = $account_service->get_cached_account_data();
+		$account_service->set_payments_api_client( $original_api_client );
+		if ( [] === $account ) {
```

Comment:
From 991-gh-Automattic/woocommerce-payments-server#issuecomment-889721732, looks like mobile team needs to show specific screen for user to finish setting it up. When onboarding is disabled, acount is also `[]`. Can we not return `NOACCOUNT` in this case by checking if onboarding is disabled?

```suggestion
		if ( [] === $account && ! $account_service->is_on_boarding_disabled() ) {
```

Reply:
If you agree about this extra check, we need to include this case in UT.

Reply:
Actually, in the case of onboarding is disabled, we still want to avoid returning empty array. I can see maybe we return empty `stdClass` or same response as empty array but status isn't `NOACCOUNT`. Mobile team needs to handle such case differently, maybe.

Reply:
Done in cf3ef00c2e13a2c0ac9f59a90a6f99eb85e6e6a2